src/MoBi.Engine/Sbml/MoleculeInformation.cs

```diff
@@ -21,7 +21,7 @@ public class MoleculeInformation
         /// </summary>
         public bool IsMultipleTimesInOneCompartment()
         {
-            return _container.Select(container => container.Name).Any(name => _container.Any(con => name == con.Name));
+            return _container.Any(target => _container.Any(duplicate => target != duplicate && target.Name == duplicate.Name));
```

Comment:
I am not sure this is not checked when adding... the fact is that without this change, there is always a warning in the UI after importing saying you have duplicated names because the original linq was wrong. I am just fixing the check so it actually checks that there are two different instances with the same name.